### PR TITLE
docs/build: s/libOSTree/libostree/

### PR DESCRIPTION
**Pull request by cgwalters**
I find "libOSTree" awkward to type and really to look at. Let's be nicer on
people's pinky fingers and eyes and drop it all down to lowercase.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +88,7 @@ Once you have a git clone or recursive archive, building is the
 same as almost every autotools project:
 
 ```
+git submodule update --init
```
Comment (jlebon) [inline README.md]: `autogen.sh` takes care of that for us, no? Or is there something subtle going on here?

**Comment by jlebon** [comment]
LGTM, just had one minor question, but feel free to `r=jlebon` if I'm missing something.

File: README.md (hunk 1 above)
Comment (cgwalters) [inline README.md]: It does, but I think it's worth calling out explicitly that we use submodules.

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 2508814

**Pull request closed without merging**